rskj-core/src/main/java/co/rsk/peg/PegUtilsLegacy.java

```diff
@@ -0,0 +1,393 @@
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static org.ethereum.config.blockchain.upgrades.ConsensusRule.RSKIP293;
+
+/**
+ * @deprecated Methods included in this class are to be used only prior to the latest HF activation
+ * Use instead methods in {@link co.rsk.peg.PegUtils}
+ */
+@Deprecated
+public class PegUtilsLegacy {
+
+    private static final Logger logger = LoggerFactory.getLogger(PegUtilsLegacy.class);
+
+    private PegUtilsLegacy() {}
+
+    /**
+     * Legacy version for identifying if a tx is a pegout
+     * Use instead {@link co.rsk.peg.PegUtils#getTransactionType}
+     *
+     * @param tx
+     * @param federations
+     * @param activations
+     * @return true if it is a peg-out. Otherwise, returns false.
+     */
+    @Deprecated
+    public static boolean isPegOutTx(BtcTransaction tx, List<Federation> federations, ActivationConfig.ForBlock activations) {
+        return isPegOutTx(tx, activations, federations.stream().filter(Objects::nonNull).map(Federation::getStandardP2SHScript).toArray(Script[]::new));
+    }
+
+    /**
+     * Legacy version for identifying if a tx is a pegout
+     * Use instead {@link co.rsk.peg.PegUtils#getTransactionType}
+     *
+     * @param btcTx
+     * @param activations
+     * @param fedStandardP2shScripts
+     * @return true if it is a peg-out. Otherwise, returns false.
+     */
+    @Deprecated
+    protected static boolean isPegOutTx(BtcTransaction btcTx, ActivationConfig.ForBlock activations, Script ... fedStandardP2shScripts) {
+        int inputsSize = btcTx.getInputs().size();
+        for (int i = 0; i < inputsSize; i++) {
+            TransactionInput txInput = btcTx.getInput(i);
+            Optional<Script> redeemScriptOptional = BitcoinUtils.extractRedeemScriptFromInput(btcTx.getInput(i));
+            if (!redeemScriptOptional.isPresent()) {
+                continue;
+            }
+
+            Script redeemScript = redeemScriptOptional.get();
+            if (activations.isActive(ConsensusRule.RSKIP201)) {
+                // Extract standard redeem script since the registered utxo could be from a fast bridge or erp federation
+                RedeemScriptParser redeemScriptParser = RedeemScriptParserFactory.get(txInput.getScriptSig().getChunks());
+                try {
+                    redeemScript = redeemScriptParser.extractStandardRedeemScript();
+                } catch (ScriptException e) {
+                    // There is no redeem script
+                    continue;
+                }
+            }
+
+            Script outputScript = ScriptBuilder.createP2SHOutputScript(redeemScript);
+            if (Arrays.stream(fedStandardP2shScripts).anyMatch(federationPayScript -> federationPayScript.equals(outputScript))) {
+                return true;
+            }
```

Comment:
```suggestion
            if (redeemScriptOptional.isPresent()) {
              Script redeemScript = redeemScriptOptional.get();
              if (activations.isActive(ConsensusRule.RSKIP201)) {
                  // Extract standard redeem script since the registered utxo could be from a fast bridge or erp federation
                  RedeemScriptParser redeemScriptParser = RedeemScriptParserFactory.get(txInput.getScriptSig().getChunks());
                  try {
                      redeemScript = redeemScriptParser.extractStandardRedeemScript();
                  } catch (ScriptException e) {
                      // There is no redeem script
                      continue;
                  }
              }
  
              Script outputScript = ScriptBuilder.createP2SHOutputScript(redeemScript);
              if (Arrays.stream(fedStandardP2shScripts).anyMatch(federationPayScript -> federationPayScript.equals(outputScript))) {
                  return true;
              }
          }
```

Reply:
Cyclomatic complexity is when you have nested ifs. What I'm proposing is the opposite, to have independent if instead.